### PR TITLE
[Angular] Add eventManager event unsubscriptions

### LINKED ISSUE #10153: [Angular] user-management.component: eventManager subscription is not destroyed
##### **JHipster Version(s)**

```
jal-bo@0.0.0 ****
`-- generator-jhipster@6.1.2

```

##### **JHipster configuration, a `.yo-rc.json` file generated in the root folder**

<details>
<summary>.yo-rc.json file</summary>
<pre>
{
  "generator-jhipster": {
    "promptValues": {
      "packageName": "****",
      "nativeLanguage": "en"
    },
    "jhipsterVersion": "6.1.2",
    "applicationType": "monolith",
    "baseName": "****",
    "packageName": "****",
    "packageFolder": "***",
    "serverPort": "8080",
    "authenticationType": "jwt",
    "cacheProvider": "ehcache",
    "enableHibernateCache": true,
    "websocket": false,
    "databaseType": "sql",
    "devDatabaseType": "h2Disk",
    "prodDatabaseType": "mariadb",
    "searchEngine": false,
    "messageBroker": false,
    "serviceDiscoveryType": false,
    "buildTool": "maven",
    "enableSwaggerCodegen": false,
    "jwtSecretKey": "bXktc2VjcmV0LXRva2VuLXRvLWNoYW5nZS1pbi1wcm9kdWN0aW9uLWFuZC10by1rZWVwLWluLWEtc2VjdXJlLXBsYWNl",
    "useSass": true,
    "clientPackageManager": "npm",
    "clientFramework": "angularX",
    "clientTheme": "none",
    "clientThemeVariant": "",
    "testFrameworks": ["protractor"],
    "jhiPrefix": "jhi",
    "entitySuffix": "",
    "dtoSuffix": "DTO",
    "otherModules": [],
    "enableTranslation": true,
    "nativeLanguage": "en",
    "languages": ["en", "fr"]
  }
}

</pre>
</details>

##### **Environment and Tools**

openjdk version "1.8.0_212"
OpenJDK Runtime Environment (AdoptOpenJDK)(build 1.8.0_212-b04)
OpenJDK 64-Bit Server VM (AdoptOpenJDK)(build 25.212-b04, mixed mode)

git version 2.20.1.windows.1

node: v10.15.0

npm: 6.8.0

yeoman: 2.0.5

yarn: 1.13.0

Docker version 18.09.2, build 6247962

docker-compose version 1.23.2, build 1110ad01

##### **Overview of the issue**

When you are on the user-component list, as the eventManager subscription is not destroyed when the component is, the user-service request (loadAll -> this.userService.query) is made N time (N is the number of the user-management.component is created.

##### **Motivation for or Use Case**

It's important to unsubscribe subcriptions, other than http and routing that Angular handles, because it leads to memory leak and lower performances

##### **Reproduce the error**

1. Go to the user-management page (user-management.component)
2. Create a user
3. Click on "Edit" for this user and cancel to go back to the list
4. Open your developer tools -> got to network tab
5. Remove the created user
6. See the number of requests made by the subscription
##### **Suggest a Fix**

What is done in the component
```javascript
registerChangeInUsers() {
    this.eventManager.subscribe('userListModification', response => this.loadAll());
  }
```
It could be destroyed in the ngOnDestroy method of the component:

```javascript
private _destroyed$ = new Subject();
....
registerChangeInUsers() {
    this.eventManager
         .pipe(
              takeUntil(this._destroyed$)
         )
         .subscribe('userListModification', response => this.loadAll());
  }
....
public ngOnDestroy (): void {
    this._destroyed$.next();
    this._destroyed$.complete();
}
```
##### **JHipster Version(s)**

6.1.2, don't know.

##### **Browsers and Operating System**

Windows 10, all browsers
-   [x] Checking this box is mandatory (this is just to show you read everything)

### PR DESCRIPTION
**Pull request by QuentinLeGoff**
In user-management and home component, events handled by eventManager are not unsubscribed. It causes anormal behavior as requesting multiple times the API for a resource for example.

Fix #10153 

-   Please make sure the below checklist is followed for Pull Requests.

-   [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
-   [ ] Tests are added where necessary
-   [ ] Documentation is added/updated where necessary
-   [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by pascalgrimaud** [comment]
Thanks and congrats for your 1st contribution @QuentinLeGoff

I'll let @vishal423 to review and merge your PR, as I'm not a front expert :p

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs [hunk 1]
```diff
@@ -41,8 +42,9 @@ export class UserMgmtComponent implements OnInit, OnDestroy {
     users: User[];
     error: any;
     success: any;
+    eventSubscriber: Subscription;
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs]: good to rename variable to `userListSubscription`

File: generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs [hunk 2]
```diff
@@ -87,10 +89,11 @@ export class UserMgmtComponent implements OnInit, OnDestroy {
         <%_ if (databaseType !== 'cassandra') { _%>
         this.routeData.unsubscribe();
         <%_ } _%>
+        this.eventManager.destroy(this.eventSubscriber);
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/admin/user-management/user-management.component.ts.ejs]: would be safe to add a null check on `this.eventSubscriber`

File: generators/client/templates/angular/src/main/webapp/app/home/home.component.ts.ejs [hunk 3]
```diff
@@ -32,9 +33,10 @@ import { <% if (authenticationType !== 'oauth2') { %>LoginModalService<% } else
     ]
 
 })
-export class HomeComponent implements OnInit {
+export class HomeComponent implements OnInit <%_ if (authenticationType !== 'oauth2') { _%>, OnDestroy <%_ } _%> {
     account: Account;
     <%_ if (authenticationType !== 'oauth2') { _%>
+    eventSubscriber: Subscription;
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/home/home.component.ts.ejs]: would be good to rename variable to `authSubscription`

File: generators/client/templates/angular/src/main/webapp/app/home/home.component.ts.ejs [hunk 4]
```diff
@@ -77,4 +79,10 @@ export class HomeComponent implements OnInit {
         this.loginService.login();
         <%_ }_%>
     }
+
+    <%_ if (authenticationType !== 'oauth2') { _%>
+    ngOnDestroy() {
+        this.eventManager.destroy(this.eventSubscriber);
```
Comment (vishal423) [inline generators/client/templates/angular/src/main/webapp/app/home/home.component.ts.ejs]: would be safe to add a null check on `this.eventSubscriber`

**Comment by QuentinLeGoff** [comment]
PR claimed with https://opencollective.com/generator-jhipster/expenses/9456

**Comment by pmverma** [comment]
Congrats for your 1st contribution @QuentinLeGoff.

**Pull request merged**